plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive2OnDataLakeOverProxy.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.plugin.hive.containers.HiveHadoop;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.containers.Httpd;
+import io.trino.testing.containers.Minio;
+import org.testcontainers.containers.Network;
+
+import java.util.Map;
+
+import static io.trino.testing.containers.Httpd.builderForHttpProxy;
+
+public class TestHive2OnDataLakeOverProxy
+        extends BaseTestHiveOnDataLake
+{
+    public TestHive2OnDataLakeOverProxy()
+    {
+        super(HiveHadoop.DEFAULT_IMAGE);
+    }
+
+    @Override
+    protected DistributedQueryRunner createQueryRunner(Map<String, String> additionalHiveProperties, Network network)
```

Comment:
The multi-stage creation of `createQueryRunner` which hops from parent to child and back to parent is tricky. Is there a way to express this more simply?

Reply:
I already spend some time trying to not replicate too much code and this was the outcome. But this is right it's kinda tricky. I will try to find a way to make it easier.